### PR TITLE
daemon: refine access testing

### PR DESCRIPTION
**Pull request by pedronis**
This is a follow-up to #9043 that tries to clarify and complete the testing of access check code and choices.

### CODE REVIEW

**Comment by pedronis** [comment]
@jhenstridge this is the follow-up I mentioned to your #9043

**Review by jhenstridge — APPROVED** [review]
Looking over the diff via https://github.com/pedronis/snappy/compare/10f22a2a23...daemon-refine-access-testing this all looks pretty good.

One small issue: we've got constants for the different polkit action IDs currently in use, but you've used string constants in the tests.  Would it make sense to export and use the named constants in the tests too? I can also see some potential value in checking the actual values as  they should match up with the policy file, so it could go either way.

**Comment by pedronis** [comment]
> I can also see some potential value in checking the actual values as they should match up with the policy file, so it could go either way.

we had situation where we used constant both in code and tests and we ended testing a typo against itself. It can still happen even in this style if the typo is common but can help, so that's why I would keep it this way.

**Review by bboozzoo — APPROVED** [review]
Last 3 commits look good to me.

**Review by stolowski — APPROVED** [review]
The test changes look good to me.

File: daemon/api_asserts_test.go [hunk 1]
```diff
@@ -55,18 +52,7 @@ func (s *assertsSuite) SetUpTest(c *check.C) {
...
-	st.Lock()
-	defer st.Unlock()
-	u, err := auth.NewUser(st, "username", "email@test.com", "macaroon", []string{"discharge"})
-	c.Assert(err, check.IsNil)
-	s.authUser = u
-}
-
-func (s *assertsSuite) userAuth(req *http.Request) {
-	req.Header.Set("Authorization", fmt.Sprintf(`Macaroon root="%s"`, s.authUser.Macaroon))
+	s.daemonWithStore(c, s)
```
Comment (mvo5) [inline daemon/api_asserts_test.go:55]: Nice

**Review by mvo5 — APPROVED** [review]

**Pull request merged**